ocaml/idl/datamodel.ml

```diff
@@ -8638,6 +8649,104 @@ let vgpu_type =
 		]
 	()
 
+module PVS_farm = struct
+	let lifecycle = [Prototyped, rel_dundee_plus, ""]
+	let refset x  = Set(Ref x)
+
+	let introduce = call
+		~name:"introduce"
+		~doc:"Introduce new PVS farm"
+		~result:(Ref _pvs_farm, "the new PVS farm")
+		~params:
+		[ String,"name","name of the PVS farm"
+		]
+		~lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let forget = call
+		~name:"forget"
+		~doc:"Remove a farm's meta data"
+		~params:
+		[ Ref _pvs_farm, "self", "this PVS farm"
+		]
+		~lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+
+	let set_name = call
+		~name:"set_name"
+		~doc:"Update the name of the PVS farm"
+		~params:
+		[ Ref _pvs_farm, "self", "this PVS farm"
+		; String, "value", "name to be used"
+		]
+		~lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let add_cache_storage = call
+		~name:"add_cache_storage"
+		~doc:"Add a cache SR for the proxies on the farm"
+		~params:
+		[ Ref _pvs_farm, "self", "this PVS farm"
+		; Ref _sr, "value", "SR to be used"
+		]
+		~lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let remove_cache_storage = call
+		~name:"remove_cache_storage"
+		~doc:"Remove a cache SR for the proxies on the farm"
+		~params:
+		[ Ref _pvs_farm, "self", "this PVS farm"
+		; Ref _sr, "value", "SR to be removed"
+		]
+		~lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let obj =
+		let null_str = Some (VString "") in
+		let null_set = Some (VSet []) in
+		create_obj
+		~name: _pvs_farm
+		~descr:"machines serving blocks of data for provisioning VMs"
+		~doccomments:[]
+		~gen_constructor_destructor:false
+		~gen_events:true
+		~in_db:true
+		~lifecycle
+		~persist:PersistEverything
+		~in_oss_since:None
+		~messages_default_allowed_roles:_R_POOL_OP
+		~contents:
+		[ uid     _pvs_farm ~lifecycle
+
+		; field   ~qualifier:StaticRO ~lifecycle
+		          ~ty:String "name" ~default_value:null_str
+		          "Name of the PVS farm. Must match name configured in PVS"
+
+		; field   ~qualifier:StaticRO ~lifecycle
```

Comment:
We should make this `DynamicRO` (it was wrong in the design; I will update that).